File: lib_arch/src/main/java/com/yyxnb/arch/common/Bus.java
package com.yyxnb.arch.common;

import android.arch.lifecycle.LifecycleOwner;
import android.arch.lifecycle.Observer;
import android.support.annotation.NonNull;

import com.jeremyliao.liveeventbus.LiveEventBus;

import static com.yyxnb.arch.common.ArchConfig.MSG_EVENT;

/**
 * 方便MsgEvent的使用
 */
public class Bus {

    public static void post(@NonNull MsgEvent msgEvent) {
        LiveEventBus.get(MSG_EVENT, MsgEvent.class).post(msgEvent);
    }

    public static void post(@NonNull MsgEvent msgEvent, long delay) {
        LiveEventBus.get(MSG_EVENT, MsgEvent.class).postDelay(msgEvent, delay);
    }

    public static void broadcast(@NonNull MsgEvent msgEvent) {
        LiveEventBus.get(MSG_EVENT, MsgEvent.class).broadcast(msgEvent);
    }

    public static void observe(@NonNull LifecycleOwner owner, @NonNull Observer<MsgEvent> observer) {
        LiveEventBus.get(MSG_EVENT, MsgEvent.class).observe(owner, observer);
    }

    public static void observeSticky(@NonNull LifecycleOwner owner, @NonNull Observer<MsgEvent> observer) {
        LiveEventBus.get(MSG_EVENT, MsgEvent.class).observeSticky(owner, observer);
    }
}

File: lib_arch/src/main/java/com/yyxnb/arch/common/Bus.kt
package com.yyxnb.arch.common

import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.Observer
import com.jeremyliao.liveeventbus.LiveEventBus

/**
 * 方便MsgEvent的使用
 */
object Bus {
    fun post(msgEvent: MsgEvent) {
        LiveEventBus.get(ArchConfig.MSG_EVENT, MsgEvent::class.java).post(msgEvent)
    }

    @JvmStatic
    fun post(msgEvent: MsgEvent, delay: Long) {
        LiveEventBus.get(ArchConfig.MSG_EVENT, MsgEvent::class.java).postDelay(msgEvent, delay)
    }

    fun broadcast(msgEvent: MsgEvent) {
        LiveEventBus.get(ArchConfig.MSG_EVENT, MsgEvent::class.java).broadcast(msgEvent, false, false)
    }

    @JvmStatic
    fun observe(owner: LifecycleOwner, observer: Observer<MsgEvent>) {
        LiveEventBus.get(ArchConfig.MSG_EVENT, MsgEvent::class.java).observe(owner, observer)
    }

    fun observeSticky(owner: LifecycleOwner, observer: Observer<MsgEvent>) {
        LiveEventBus.get(ArchConfig.MSG_EVENT, MsgEvent::class.java).observeSticky(owner, observer)
    }
}